package/network/services/omcproxy/files/omcproxy.init

```diff
@@ -1,62 +1,88 @@
 #!/bin/sh /etc/rc.common
-# Copyright (C) 2010-2014 OpenWrt.org
+# Copyright (C) 2018 OpenWrt.org
 
 START=99
 USE_PROCD=1
 PROG=/usr/sbin/omcproxy
 
-# Uncomment to enable verbosity 
-#OPTIONS="-v" 
+# Uncomment to enable verbosity
+#OPTIONS="-v"
 PROXIES=""
 
-
 omcproxy_add_proxy() {
-	local uplink downlink scope proxy
-	config_get uplink $1 uplink
-	config_get downlink $1 downlink
+	local proxy scope uplink downlinks device up
+
 	config_get scope $1 scope
+	config_get uplink $1 uplink
+	config_get downlinks $1 downlink
+
+	[ -n "$uplink" ] || return
+
+	local status="$(ubus -S call "network.interface.$uplink" status)"
+	[ -n "$status" ] || return
+
+	json_load "$status"
+	json_get_var device l3_device
+	json_get_var up up
+
+	[ -n "$device" -a "$up" = "1" ] || {
+		procd_append_param error "$uplink is not up"
+		return;
+	}
+
+	proxy="$device"
+
+	for downlink in $downlinks; do
+		local status="$(ubus -S call "network.interface.$downlink" status)"
```

Comment:
As mentioned above there's no need to use ubus network calls directly if you use the network.sh wrapper functions